flake: TestAPIExportEndpointSliceWithPartitionPrivate

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2733/pull-ci-kcp-dev-kcp-main-e2e-sharded/1621190822823202816

```
=== RUN   TestAPIExportEndpointSliceWithPartitionPrivate
=== PAUSE TestAPIExportEndpointSliceWithPartitionPrivate
=== CONT  TestAPIExportEndpointSliceWithPartitionPrivate
=== CONT  TestAPIExportEndpointSliceWithPartitionPrivate
    apiexportendpointslice_test.go:166: Saving test artifacts for test "TestAPIExportEndpointSliceWithPartitionPrivate" under "/logs/artifacts/TestAPIExportEndpointSliceWithPartitionPrivate/850502309".
    apiexportendpointslice_test.go:166: Starting kcp servers...
    kcp.go:512: running: /go/src/github.com/kcp-dev/kcp/bin/kcp start --root-directory /tmp/TestAPIExportEndpointSliceWithPartitionPrivate1625203280/002/kcp/main --secure-port=35403 --embedded-etcd-client-port=33973 --embedded-etcd-peer-port=45529 --embedded-etcd-wal-size-bytes=5000 --kubeconfig-path=/tmp/TestAPIExportEndpointSliceWithPartitionPrivate1625203280/002/kcp/main/admin.kubeconfig --feature-gates=CustomResourceValidationExpressions=true --audit-log-path /logs/artifacts/TestAPIExportEndpointSliceWithPartitionPrivate/850502309/artifacts/kcp/main/kcp.audit --audit-policy-file /tmp/TestAPIExportEndpointSliceWithPartitionPrivate1625203280/001/audit-policy.yaml
=== CONT  TestAPIExportEndpointSliceWithPartitionPrivate
    kcp.go:986: waiting for readiness for server at https://10.130.86.12:35403
    kcp.go:1029: success contacting https://10.130.86.12:35403/readyz
    kcp.go:1029: success contacting https://10.130.86.12:35403/livez
    kcp.go:1007: server at https://10.130.86.12:35403 is ready
    apiexportendpointslice_test.go:166: Started kcp servers after 21.76018484s
    apiexportendpointslice_test.go:169: Created root:organization workspace root:e2e-workspace-jw5jf as /clusters/2k5ag0ea4jha8byv on shard "root"
    apiexportendpointslice_test.go:170: Created root:universal workspace root:e2e-workspace-jw5jf:e2e-workspace-c8sl9 as /clusters/32ga6phras9jkc1u on shard "root"
    apiexportendpointslice_test.go:171: Created root:universal workspace root:e2e-workspace-jw5jf:e2e-workspace-fffph as /clusters/8836fy5pme4uvgxg on shard "root"
    apiexportendpointslice_test.go:210: Creating the APIExport
    apiexportendpointslice_test.go:215: Creating the APIExportEndpointSlice
    apiexportendpointslice_test.go:218: 
        	Error Trace:	apiexportendpointslice_test.go:218
        	Error:      	Received unexpected error:
        	            	apiexportendpointslices.apis.kcp.io "my-slicek5rht" is forbidden: unable to create APIExportEndpointSlice: no permission to bind to export root:e2e-workspace-jw5jf:e2e-workspace-c8sl9:my-export
        	Test:       	TestAPIExportEndpointSliceWithPartitionPrivate
    kcp.go:499: cleanup: canceling context
    kcp.go:503: cleanup: waiting for shutdownComplete
    kcp.go:507: cleanup: received shutdownComplete
--- FAIL: TestAPIExportEndpointSliceWithPartitionPrivate (24.70s)
```

Probably needs an Eventually?

--- a/test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go
+++ b/test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go
@@ -214,8 +214,14 @@ func TestAPIExportEndpointSliceWithPartitionPrivate(t *testing.T) {
 
 	t.Logf("Creating the APIExportEndpointSlice")
 	sliceClient := kcpClusterClient.ApisV1alpha1().APIExportEndpointSlices()
-	slice, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
-	require.NoError(t, err)
+	// allow some time for APIExport to be synced onto the cache server
+	framework.Eventually(t, func() (bool, string) {
+		slice, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+		if err != nil {
+			return false, err.Error()
+		}
+		return true, ""
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected successful creation of APIExportEndpointSlice")
 	sliceName := slice.Name
 
 	framework.Eventually(t, func() (bool, string) {

